crates/pubsub/src/service.rs

```diff
@@ -89,15 +91,13 @@ where
         // Re-subscribe to all active subscriptions
         tracing::debug!(count = self.subs.len(), "Re-starting active subscriptions");
 
-        // Drop all server IDs. We'll re-insert them as we get responses.
-        self.subs.drop_server_ids();
         // Dispatch all subscription requests
         self.subs
             .iter()
-            .map(|(_, sub)| sub.request().serialized().to_owned())
+            .map(|(_, sub)| sub.request().to_owned())
             .collect::<Vec<_>>()
             .into_iter()
-            .try_for_each(|brv| self.dispatch_request(brv))?;
+            .try_for_each(|req| self.dispatch_subscription(req))?;
```

Comment:
What's the reason for collecting here? cc @prestwich

Reply:
I think that without collecting closure requires unique access to `*self` but it is already borrowed

Reply:
yes, self is borrowed by the `Map<Iter<..>>` and collecting ends that borrow so that the `try_for_each` can get `&mut self`

Reply:
could probably be simplified into a single `try_for_each`

Reply:
If `dispatch_subscription` is inlined it can be because the borrows get split to the individual fields